examples/tls-client.rs

```diff
@@ -0,0 +1,122 @@
+// SPDX-FileCopyrightText: Copyright (c) 2017-2023 slowtec GmbH <post@slowtec.de>
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+//! Asynchronous TLS client example
+use tokio::net::TcpStream;
+
+use std::{
+    fs::File,
+    io::{self, BufReader},
+    net::SocketAddr,
+    path::Path,
+    sync::Arc,
+};
+
+use pkcs8::der::Decode;
+use rustls_pemfile::{certs, pkcs8_private_keys};
+use tokio_rustls::rustls::{self, Certificate, OwnedTrustAnchor, PrivateKey};
+use tokio_rustls::{webpki, TlsConnector};
+
+fn load_certs(path: &Path) -> io::Result<Vec<Certificate>> {
+    certs(&mut BufReader::new(File::open(path)?))
+        .map_err(|_| io::Error::new(io::ErrorKind::InvalidInput, "invalid cert"))
+        .map(|mut certs| certs.drain(..).map(Certificate).collect())
```

Comment:
Why not `certs.into_iter()`?

Reply:
Most of this code for load certs and keys is from an example of the official git of the tokio-rustls project.
[https://github.com/tokio-rs/tls/blob/master/tokio-rustls/examples/server/src/main.rs](https://github.com/tokio-rs/tls/blob/master/tokio-rustls/examples/server/src/main.rs).

So, I believe we can leave it like this.

Reply:
If you copied code than please at least add a comment with a link to it. Since the original source is only an example and the licenses are compatible I don't expect any complaints. Nevertheless respecting copyrights is always required.

Reply:
I've never seen anyone consider copying code without respecting copyright when using code that exemplifies how to use a library from another project. But if you consider that this is necessary, I see no problem in making this note.